crates/breez-sdk/core/src/persist/sqlite.rs

```diff
@@ -133,6 +134,35 @@ impl SqliteStorage {
               PRIMARY KEY (deposit_tx_id, deposit_vout)              
             );",
             "ALTER TABLE payment_metadata ADD COLUMN lnurl_description TEXT;",
+            "
+            ALTER TABLE payments ADD COLUMN withdraw_tx_id TEXT;
+            ALTER TABLE payments ADD COLUMN deposit_tx_id TEXT;
+            ALTER TABLE payments ADD COLUMN spark INTEGER;
+            CREATE TABLE payment_details_lightning (
+              payment_id TEXT PRIMARY KEY,
+              invoice TEXT NOT NULL,
+              payment_hash TEXT NOT NULL,
+              destination_pubkey TEXT NOT NULL,
+              description TEXT,
+              preimage TEXT,
+              FOREIGN KEY (payment_id) REFERENCES payments(id) ON DELETE CASCADE
+            );
```

Comment:
Seems like only the lightning details are saved in another table. If this is because we only have one field for other kind of payments then I think it will change in the future and better to be consistent.
I am OK also with use one table with nullable fields for all kind of payments.

Reply:
It is because there is only one field, yes. Ideally everything is just in the same table. The lightning details are extracted out into their own table, so that fields don't have to be nullable. This makes reasoning about the data structure easier to understand, because you never have to consider a null invoice but a non-null payment hash for example.

I think we should move the withdraw/deposit/spark to another table when it is needed from that nullability perspective. So if you add a field basically.